### PR TITLE
Adding slxos_lldp module

### PR DESCRIPTION
**Pull request by bigmstone**
##### SUMMARY
This PR adds the `slxos_lldp` module

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
`slxos_lldp`

##### ANSIBLE VERSION
```
ansible 2.6.0 (slxos-lldp 43342bf3d1) last updated 2018/04/24 17:49:06 (GMT -500)
  config file = None
  configured module search path = [u'/Users/mstone/dev/stackstorm/ansible/library']
  ansible python module location = /Users/mstone/dev/stackstorm/ansible/lib/ansible
  executable location = /Users/mstone/dev/stackstorm/ansible/bin/ansible
  python version = 2.7.14 (default, Mar 29 2018, 15:58:25) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### ADDITIONAL INFORMATION
```
⟩ ansible-playbook --ask-pass -i hosts.yml lldp.yml
SSH password:
ERROR! no action detected in task. This often indicates a misspelled module name, or incorrect module path.

The error appears to have been in '/Users/mstone/dev/stackstorm/ansible-slxos/lldp.yml': line 7, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - name: LLDP
      ^ here
```
```
⟩ ansible-playbook --ask-pass -i hosts.yml lldp.yml
SSH password:

PLAY [slx] *********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

TASK [LLDP] ********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
ok: [dc2leaf2]

PLAY RECAP *********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
dc2leaf1                   : ok=1    changed=1    unreachable=0    failed=0
```

### CODE REVIEW

**Comment by bigmstone** [comment]
@gundalow This PR is ready for merge.

**Comment by bigmstone** [comment]
@gundalow This PR is ready for merge.

**Review by LindsayHill — APPROVED** [review]

**Comment by gundalow** [comment]
@bigmstone Can you rebase please

File: lib/ansible/modules/network/slxos/slxos_lldp.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    result = {'changed': False}
+
+    if warnings:
+        result['warnings'] = warnings
+
+    commands = []
+
+    print(HAS_LLDP)
+    print(module.params['state'])
```
Comment (ganeshrn) [inline lib/ansible/modules/network/slxos/slxos_lldp.py]: Please remove `print` lines

File: lib/ansible/modules/network/slxos/slxos_lldp.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    if module.params['state'] == 'absent' and HAS_LLDP:
+        print('Nope')
```
Comment (ganeshrn) [inline lib/ansible/modules/network/slxos/slxos_lldp.py]: same as above

File: lib/ansible/modules/network/slxos/slxos_lldp.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        commands.append('protocol lldp')
+        commands.append('disable')
+    elif module.params['state'] == 'present' and not HAS_LLDP:
+        print('Adding the commands, captin')
```
Comment (ganeshrn) [inline lib/ansible/modules/network/slxos/slxos_lldp.py]: same as above

File: lib/ansible/modules/network/slxos/slxos_lldp.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
```
Comment (ganeshrn) [inline lib/ansible/modules/network/slxos/slxos_lldp.py]: This should be
```
'supported_by': 'community'
```

**Review by ganeshrn — CHANGES_REQUESTED** [review]

**Comment by bigmstone** [comment]
@ganeshrn print statements removed
@gundalow conflicts resolved

**Comment by gundalow** [comment]
Hi, 
See if http://docs.ansible.com/ansible/devel/dev_guide/index.html helps with the rebase, thanks in advance.

**Comment by gundalow** [comment]
@bigmstone Can you please rebase, the merge deadline for new functionality in 2.6 is 2018-05-25 Thanks in advance

**Comment by bigmstone** [comment]
@gundalow Should be fixed up, thanks.

**Comment by bigmstone** [comment]
@gundalow @ganeshrn Since this is ready to go and the deadline is today can we get this pushed through?

**Review by ganeshrn — APPROVED** [review]

**Comment by gundalow** [comment]
rebuild_merge

**Comment by gundalow** [comment]
Merged into `devel` for release in Ansible 2.7

**Pull request merged**